environment/k8s_environment.go

```diff
@@ -158,6 +163,9 @@ func (env *k8sEnvironment) GetServiceDetails(remotePort uint16) (*ServiceDetails
 // TearDown cycles through all the specifications and tears down the deployments. This typically entails cleaning
 // up port forwarding requests and deleting the namespace that then destroys all definitions.
 func (env *k8sEnvironment) TearDown() {
+	if ginkgo.CurrentGinkgoTestDescription().Failed { // If a test fails, dump logs
```

Comment:
To avoid needing to import Ginkgo in environments, split out this function. Have a new `ExportLogsToFolder(folder string, logPrefix log.Buffer)` function in the environment interface, that way Ginkgo isn't needed and the logic to whether logs are needed to be written is still handled in the `actions` package.

To replicate the functionality where you output a heading to the log file with more info, you can then instantiate the buffer in teardown with that output and simply copy the buffer each time and append to it.

Reply:
Followed this, but went a little simpler with it, will see how you like it.